STM32F405RGTx/Core/Inc/measurements_thread.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include "util.hpp"
+
+#define ADC_NUM_CONVERSIONS         6U
+#define ADC_DECIMATION_COEFF        256U
+
+class MeasurementsThread {
+    public:
+        static void initialize();
+
+        static void stopMeasurements();
+        static void resumeMeasurements();
+    private:
+        static RTOSThread thread;
+        static void runMeasurements(void* args);
+
+        static uint16_t ADC_buffer[ADC_NUM_CONVERSIONS*ADC_DECIMATION_COEFF];
+
+        static void processData();
+        
+        static void startADCandDMA();
+
+        static void HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef* hadc);
```

Comment:
The HAL_ADC_ConvCpltCallback is an ISR, so it cannot be declared like this. It needs to be declared in global namespace so it can be linked with the function call in the ISR.